src/app/pages/StakingPage/Features/DelegationList/index.tsx

```diff
@@ -24,15 +24,23 @@ import { DelegationItem } from './DelegationItem'
 
 interface Props {
   type: 'active' | 'debonding'
-  delegations: Delegation[]
+  delegations: Delegation[] | DebondingDelegation[]
 }
 
 /**
  * Renders the list of delegations
  */
 export const DelegationList = memo((props: Props) => {
   const type = props.type
-  const delegations = props.delegations
+
+  // Create a combined unique key because validatorAddress is not unique in debondings
```

Comment:
Is validator address unique in epoch ?

Reply:
yes https://github.com/oasisprotocol/oasis-core/blob/v22.1.10/go/consensus/tendermint/apps/staking/transactions.go#L445

Reply:
Yes - I checked on oasisscan API and gRPC. I haven't been able to confirm on oasisindexer yet

Reply:
Update: oasis-indexer does not merge debondings in the same epoch
https://index.oasislabs.com/v1/consensus/accounts/oasis1qz78ap0456g2rk7j6rmtvasc9v2kjhz2s58qgj90/debonding_delegations
```json
    {
      "amount": 999999999,
      "shares": 999999999,
      "address": "oasis1qpn83e8hm3gdhvpfv66xj3qsetkj3ulmkugmmxn3",
      "debond_end": 16657
    },
    {
      "amount": 999999999,
      "shares": 999999999,
      "address": "oasis1qpn83e8hm3gdhvpfv66xj3qsetkj3ulmkugmmxn3",
      "debond_end": 16657
    },
    {
      "amount": 999999999,
      "shares": 999999999,
      "address": "oasis1qpn83e8hm3gdhvpfv66xj3qsetkj3ulmkugmmxn3",
      "debond_end": 16658
    }
```

Solutions:
- merge them on the frontend
- use array index as key

Reply:
how do they get this data?
and how do they maintain it? :monocle_face:

Reply:
@aefhm

Reply:
> how do they get this data?

We read every [staking](https://github.com/oasisprotocol/oasis-indexer/blob/8a306b8eb65f9244b13462a5d9ede7cafaa2614c/storage/oasis/client.go#L292) backend [event](https://github.com/oasisprotocol/oasis-core/blob/3fa509673fcf9de8db239db2fdef0081b1abb8b2/go/consensus/tendermint/apps/staking/transactions.go#L471-L478).

> how do they maintain it

Uhhh, 😓 they are stored as separate rows/records in DB and theoretically deleted on reclaim.

Reply:
@lukaw3d I take it that we prefer them merged?

Reply:
This preference only comes from a react implementation detail, so it shouldn't influence API design. One reason to merge them would be to be consistent with gRPC :shrug: 

Otherwise I actually like that they are separate